core/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -569,6 +562,23 @@ public static void generateFailureXContent(XContentBuilder builder, Params param
     }
 
     /**
+     * Parses the output of {@link #generateFailureXContent(XContentBuilder, Params, Exception, boolean)}
+     */
+    public static ElasticsearchException failureFromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.nextToken();
+        ensureFieldName(parser, token, ERROR);
+
+        token = parser.nextToken();
+        if (token.isValue() || token == XContentParser.Token.VALUE_NULL) {
+            return new ElasticsearchException(buildMessage("exception", parser.textOrNull(), null));
+        }
+
+        ensureExpectedToken(token, XContentParser.Token.START_OBJECT, parser::getTokenLocation);
```

Comment:
how about the root cause? I remember we discussed that it is a problem when we get more than one, but can we add a TODO for it somewhere or do at least something when we have a single one? I don't see where we parse them but I may be missing something. Maybe they end up skipped where we parse metadata arrays in `innerFromXContent`